serde-api/src/main/java/io/micronaut/serde/config/DeserializationConfiguration.java

```diff
@@ -52,4 +53,14 @@ public interface DeserializationConfiguration {
     default boolean isFailOnNullForPrimitives() {
         return false;
     }
+
+    /**
+     * Whether the supertype is used by default when no supertype is resolved.
+     * @return True to avoid the supertype and use `defaultImpl` property
+     */
+    @NextMajorVersion("Inline to true to have the behaviour the same as for Jackson")
+    @Bindable(defaultValue = StringUtils.FALSE)
+    default boolean isAvoidSupertypeSubtype() {
```

Comment:
This method name is a bit confusing and so is the description. Can we rename it to `subtypes-require-default-impl: true` which I think is more descriptive of what it is?